server/command/command.go

```diff
@@ -87,6 +91,81 @@ func sanitizeShareName(name string) string {
 	return result
 }
 
+// syncChannelMembersToMatrixRoom creates ghost users for all channel members and joins them to the Matrix room
+func (c *Handler) syncChannelMembersToMatrixRoom(channelID, roomID string) (int, int, error) {
+	matrixClient := c.plugin.GetMatrixClient()
+	if matrixClient == nil {
+		return 0, 0, errors.New("matrix client not available")
+	}
+
+	// Get all channel members with pagination
+	var allMembers []model.ChannelMember
+	offset := 0
+	limit := 100
+
+	for {
```

Comment:
Any particular reason to use two loops here? Couldn't we use the pagination loop to create the users?